tests/pipelines/stable_diffusion_2/test_stable_diffusion_latent_upscale.py

```diff
@@ -0,0 +1,218 @@
+
+        components = {
+            "unet": model.eval(),
+            "vae": vae.eval(),
+            "scheduler": scheduler,
+            "text_encoder": text_encoder,
+            "tokenizer": tokenizer,
+        }
+
+        return components
+
+    def get_dummy_inputs(self, device, seed=0):
+        if str(device).startswith("mps"):
+            generator = torch.manual_seed(seed)
+        else:
+            generator = torch.Generator(device=device).manual_seed(seed)
+        inputs = {
+            "prompt": "A painting of a squirrel eating a burger",
+            "image": self.dummy_image.cpu(),
+            "generator": generator,
+            "num_inference_steps": 2,
+            "output_type": "numpy",
+        }
+        return inputs
+
+    def test_inference(self):
+        device = "cpu"
+
+        components = self.get_dummy_components()
+        pipe = self.pipeline_class(**components)
+        pipe.to(device)
+        pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs(device)
+        image = pipe(**inputs).images
+        image_slice = image[0, -3:, -3:, -1]
+
+        self.assertEqual(image.shape, (1, 256, 256, 3))
+        expected_slice = np.array(
+            [0.5526507, 0.61817056, 0.6035498, 0.65823257, 0.61786944, 0.64031905, 0.7637722, 0.63346744, 0.6339706]
+        )
+        max_diff = np.abs(image_slice.flatten() - expected_slice).max()
+        self.assertLessEqual(max_diff, 1e-3)
+
+    def test_inference_batch_single_identical(self):
+        self._test_inference_batch_single_identical(relax_max_difference=True)
+
+
+@require_torch_gpu
+@slow
+class StableDiffusionLatentUpscalePipelineIntegrationTests(unittest.TestCase):
+    def tearDown(self):
+        super().tearDown()
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    def test_latent_upscaler_fp16(self):
+        generator = torch.manual_seed(33)
+
+        pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", torch_dtype=torch.float16)
+        pipe.to("cuda")
+
+        upscaler = StableDiffusionLatentUpscalePipeline.from_pretrained(
+            "stabilityai/sd-x2-latent-upscaler", torch_dtype=torch.float16
+        )
+        upscaler.to("cuda")
+
+        prompt = "a photo of an astronaut high resolution, unreal engine, ultra realistic"
+
+        low_res_latents = pipe(prompt, generator=generator, output_type="latent").images
+
+        image = upscaler(
+            prompt=prompt,
+            image=low_res_latents,
+            num_inference_steps=20,
+            guidance_scale=0,
+            generator=generator,
+            output_type="np",
+        ).images[0]
+
+        expected_image = load_numpy(
+            f"https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/latent-upscaler/image_upscaled_fp16.npy"
+        )
+        assert np.abs((expected_image - image).max()) < 1e-3
+
+    def test_latent_upscaler(self):
```

Comment:
Think just one test in fp16 is enough. Could we maybe instead write a second test that only tests the latent upscaler. E.g. something based on:
```py
#!/usr/bin/env python3
import requests
from PIL import Image
from io import BytesIO
from diffusers import StableDiffusionLatentUpscalePipeline, StableDiffusionPipeline
import torch

from torchvision.transforms import functional as TF

SD_Q = 0.18215
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

model_id = "YiYiXu/latent-upscaler"
model_id = "/home/patrick_huggingface_co/latent-upscaler"

# let's download an  image
url = "https://models.rivershavewings.workers.dev/assets/sd_2x_upscaler_demo.png"
response = requests.get(url)
low_res_img = Image.open(BytesIO(response.content)).convert("RGB")

prompt = "the temple of fire by Ross Tran and Gerardo Dottori, oil on canvas"

#pipeline = StableDiffusionPipeline.from_pretrained(
#    "CompVis/stable-diffusion-v1-4", torch_dtype=torch.float16
#).to(device)
upscaler = StableDiffusionLatentUpscalePipeline.from_pretrained(
    model_id, torch_dtype=torch.float16
)
upscaler.save_pretrained("../latent-upscaler")
upscaler = upscaler.to(device)

image = TF.to_tensor(low_res_img).to(device, dtype=torch.float16) * 2 - 1
low_res_latents = upscaler.vae.encode(image.unsqueeze(0)).latent_dist.sample() * SD_Q

generator = torch.manual_seed(0)


#low_res_latents = pipeline(prompt, output_type="latent", num_inference_steps=2, generator=generator).images[0]
#with torch.no_grad():
#    image = pipeline.decode_latents(low_res_latents[None, :])
#image = pipeline.numpy_to_pil(image)
#image[0].save("../images/normal_scaled_test.png")

upscaled_image = upscaler(
    prompt=prompt,
    image=low_res_latents,
    num_inference_steps=40,
    guidance_scale=0,
    generator=generator,
).images[0]

upscaler.save_pretrained("../latent-upscaler")

upscaled_image.save("../images/upscaled_test.png")
```

Reply:
✅